internal/msgqueue/v1/postgres/exchange.go

```diff
@@ -2,15 +2,60 @@ package postgres
 
 import (
 	"context"
+	"encoding/json"
+
+	"golang.org/x/sync/errgroup"
+
+	msgqueue "github.com/hatchet-dev/hatchet/internal/msgqueue/v1"
 )
 
-func (p *PostgresMessageQueue) addTenantExchangeMessage(ctx context.Context, tenantId string, msgBytes []byte) error {
-	// determine if the exchange message is greater than 8kb
-	if len(msgBytes) > 8000 {
-		// if the message is greater than 8kb, store the message in the database
-		return p.repo.AddMessage(ctx, tenantId, msgBytes)
+func (p *PostgresMessageQueue) addTenantExchangeMessage(ctx context.Context, q msgqueue.Queue, msg *msgqueue.Message) error {
+	tenantId := msg.TenantID
+
+	if tenantId == "" {
+		return nil
+	}
+
+	err := p.RegisterTenant(ctx, tenantId)
+
+	if err != nil {
+		p.l.Error().Msgf("error registering tenant exchange: %v", err)
+		return err
+	}
+
+	queueName := msgqueue.GetTenantExchangeName(tenantId)
+
+	// if the queue name does not equal the tenant exchange name, publish the message to the queue
+	if queueName != q.Name() {
+		return p.pubNonDurableMessages(ctx, queueName, msg)
+	}
+
+	return nil
+}
+
+func (p *PostgresMessageQueue) pubNonDurableMessages(ctx context.Context, queueName string, msg *msgqueue.Message) error {
+	eg := errgroup.Group{}
+
+	for _, payload := range msg.Payloads {
+		msgCp := msg
+		msgCp.Payloads = [][]byte{payload}
+
+		msgBytes, err := json.Marshal(msgCp)
+
+		if err == nil {
+			eg.Go(func() error {
```

Comment:
possible order or race conditions running this in a go routine?

Reply:
none of our messages over the MQ depend on ordering, since we have a QOS set fairly high on RabbitMQ, so this replicates that behavior.